src/components/StudentGroupPreview.tsx

```diff
@@ -52,11 +55,12 @@ const StudentGroup: React.FC<StudentGroupProps> = ({ doc }) => {
   return (
     <StudentGroupDiv onClick={openStudentGroupHandler}>
       <Heading as="h2" size="lg">
-        {doc.studentGroupName}
+        {studentGroupName}
       </Heading>
       <ButtonDiv>
-        <Button onClick={event => deleteHandler(event)}>Delete</Button>
-        <Button>Edit</Button>
+        <IconButton icon={<DeleteIcon />} onClick={event => deleteHandler(event)} aria-label="delete">
```

Comment:
You needn't create an anonymous fn here.  i think you can just pass in the reference to the function directly

`onClick={deleteHandler}`